yarn-project/simulator/src/client/execution_result.ts

```diff
@@ -133,3 +135,18 @@ export function collectEnqueuedPublicFunctionCalls(execResult: ExecutionResult):
     ...[...execResult.nestedExecutions].flatMap(collectEnqueuedPublicFunctionCalls),
   ].sort((a, b) => b.callContext.sideEffectCounter - a.callContext.sideEffectCounter);
 }
+
+export function collectPublicTeardownFunctionCall(execResult: ExecutionResult): PublicCallRequest {
+  const teardownCalls = [
+    execResult.publicTeardownFunctionCall,
+    ...[...execResult.nestedExecutions].flatMap(collectPublicTeardownFunctionCall),
```

Comment:
`...execResult.nestedExecutions.flatMap()`. 
This is probably copied from the function above, which is also overdone 😂

Reply:
I cleaned them up a little!